### PR TITLE
Supporting fields with code domain

### PR DESCRIPTION
**Pull request by hfazai**

### CODE REVIEW

File: galite-core/src/main/kotlin/org/kopi/galite/common/FieldList.kt [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+                                   val columns: Array<ListDescription<T>>,
+                                   val autocompleteType: Int,
+                                   val autocompleteLength: Int,
+                                   val access: Boolean) {
+
+  /**
+   * Returns `true` if the list has a list action.
+   */
+  fun hasAction(): Boolean {
+    return action != null
```
Comment (ZMarouani) [inline galite-core/src/main/kotlin/org/kopi/galite/common/FieldList.kt:43]: always true

File: galite-core/src/main/kotlin/org/kopi/galite/common/FieldList.kt [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+ *
+ * @param type                the identifier of the type containing this list
+ * @param table               the statement to select data
+ * @param action              the field list action
+ * @param columns             a description of the columns
+ * @param access              true if this field is only an access to a form
+ */
+class FieldList<T: Comparable<T>?>(val type: String,
+                                   val table: Table,
+                                   val action: () -> Unit,
```
Comment (ZMarouani) [inline galite-core/src/main/kotlin/org/kopi/galite/common/FieldList.kt:33]: Should  have been nullable function

File: galite-core/src/main/kotlin/org/kopi/galite/common/FieldList.kt [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+import org.jetbrains.exposed.sql.Table
+
+/**
+ * This class represent a list of data from the database
+ *
+ * @param type                the identifier of the type containing this list
+ * @param table               the statement to select data
+ * @param action              the field list action
+ * @param columns             a description of the columns
+ * @param access              true if this field is only an access to a form
```
Comment (ZMarouani) [inline galite-core/src/main/kotlin/org/kopi/galite/common/FieldList.kt:29]: Fix parameters for func doc

File: galite-core/src/main/kotlin/org/kopi/galite/form/dsl/FormField.kt [hunk 4]
```diff
@@ -224,22 +231,59 @@ class FormField<T : Comparable<T>?>(val block: FormBlock,
   /**
    * The field model based on the field type.
    */
+  @Suppress("UNCHECKED_CAST")
```
Comment (ZMarouani) [inline galite-core/src/main/kotlin/org/kopi/galite/form/dsl/FormField.kt:234]: All casts are successful , so nothing to supress

**Review by ZMarouani — CHANGES_REQUESTED** [review]

File: galite-tests/src/test/kotlin/org/kopi/galite/tests/domain/CodeDomainTests.kt [hunk 5]
```diff
@@ -75,32 +76,14 @@ class CodeDomainTests {
     val domain = IntTestType()
 
     // test code values
-    val codes = domain.getValues()
+    val codes = (domain.type as CodeDomain<*>).codes
```
[the remark this answers is not in the document]
Comment (hfazai) [reply galite-tests/src/test/kotlin/org/kopi/galite/tests/domain/CodeDomainTests.kt:79]: to get codes

**Review by ZMarouani — APPROVED** [review]

**Pull request merged**